build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/release/BreakingChangesGenerator.java

```diff
@@ -11,56 +11,117 @@
 import com.google.common.annotations.VisibleForTesting;
 
 import org.elasticsearch.gradle.VersionProperties;
+import org.gradle.api.GradleException;
 
 import java.io.File;
 import java.io.FileWriter;
 import java.io.IOException;
 import java.nio.file.Files;
 import java.util.HashMap;
 import java.util.List;
+import java.util.Locale;
 import java.util.Map;
 import java.util.Objects;
+import java.util.Set;
 import java.util.TreeMap;
+import java.util.TreeSet;
 import java.util.stream.Collectors;
 
 import static java.util.Comparator.comparing;
 import static java.util.stream.Collectors.groupingBy;
+import static java.util.stream.Collectors.toCollection;
 
 /**
- * Generates the page that lists the breaking changes and deprecations for a minor version release.
+ * Generates the page that contains an index into the breaking changes and lists deprecations for a minor version release,
+ * and the individual pages for each breaking area.
  */
 public class BreakingChangesGenerator {
+    static void update(
+        File indexTemplateFile,
+        File indexOutputFile,
+        File outputDirectory,
+        File areaTemplateFile,
+        List<ChangelogEntry> entries
+    ) throws IOException {
+        if (outputDirectory.exists()) {
+            if (outputDirectory.isDirectory() == false) {
+                throw new GradleException("Path [" + outputDirectory + "] exists but isn't a directory!");
+            }
+        } else {
+            Files.createDirectory(outputDirectory.toPath());
+        }
 
-    static void update(File templateFile, File outputFile, List<ChangelogEntry> entries) throws IOException {
-        try (FileWriter output = new FileWriter(outputFile)) {
+        try (FileWriter output = new FileWriter(indexOutputFile)) {
             output.write(
-                generateFile(QualifiedVersion.of(VersionProperties.getElasticsearch()), Files.readString(templateFile.toPath()), entries)
+                generateIndexFile(
+                    QualifiedVersion.of(VersionProperties.getElasticsearch()),
+                    Files.readString(indexTemplateFile.toPath()),
+                    entries
+                )
             );
         }
-    }
 
-    @VisibleForTesting
-    static String generateFile(QualifiedVersion version, String template, List<ChangelogEntry> entries) throws IOException {
+        // Needs to match `changelog-schema.json`
+        final List<String> breakingAreas = List.of(
```

Comment:
Can't this be a `static` class member variable?

Reply:
Yes, I'll change that.